dag_in_context/src/interval_analysis.egg

```diff
@@ -359,3 +359,30 @@
       )
       ((set (hi-bound (Get ctx i)) (IntB hi)))
       :ruleset interval-analysis)
+
+
+(rule (
+       ;; argument has loop context
+       (Arg ty (InLoop inputs outputs))
+       ;; in the loop, the argument is passed through
+       (= (Get (Arg ty some_ctx) ith) (Get outputs (+ 1 ith)))
```

Comment:
`some_ctx` must be `(InLoop inputs outputs)`, and this expression is exactly the one you are setting lower bound of, right? We should enforce this (also makes the code look cleaner)

Reply:
Actually, `some_ctx` is probably a different context since this argument was a part of the context's inputs